### PR TITLE
Fixes based on testing.

### PR DESCRIPTION
**Pull request by JukkaSad**

### CODE REVIEW

File: game_objects/ball.js [hunk 1]
```diff
@@ -22,43 +22,39 @@ class Ball {
 	//Set the new position
 	//TODO: Apply slowing forces if not controlled
 	move(dt, closestPlayer, direction, speed) {
-
-        if (closestPlayer.position.distanceTo(this.position) < closestPlayer.size * 6) {
```
Comment (JukkaSad) [inline game_objects/ball.js:26]: What was this 6? just for testing?

File: game_objects/ball.js [hunk 2]
```diff
@@ -22,43 +22,39 @@ class Ball {
 	//Set the new position
 	//TODO: Apply slowing forces if not controlled
 	move(dt, closestPlayer, direction, speed) {
-
-        if (closestPlayer.position.distanceTo(this.position) < closestPlayer.size * 6) {
+        
+        if (closestPlayer !== null && closestPlayer.position.distanceTo(this.position) < closestPlayer.size+this.size) {
```
Comment (jaakkosadeharju) [inline game_objects/ball.js:26]: Closest player should never be null

File: game_objects/player.js [hunk 3]
```diff
@@ -78,30 +83,29 @@ class Player {
...
+        if(ball.controlledBy != this)
             return;
             
         //meters to pixels conversion factor
         let metersToPixelsRatio = 10;
-        let acceleration = force / this.ball.weight;
-        let speed = Math.sqrt(Math.pow(this.ball.speed.x, 2)+Math.pow(this.ball.speed.y, 2));
-        speed = speed + acceleration * (dt/1000) * metersToPixelsRatio;
+        let acceleration = force / ball.weight;
+        let speed = acceleration * (dt/1000) * metersToPixelsRatio;
```
Comment (jaakkosadeharju) [inline game_objects/player.js]: We could name this velocity. And everywhere where the direction is missing.

**Review by jaakkosadeharju — APPROVED** [review]

File: game_objects/ball.js (hunk 2 above)
Comment (jaakkosadeharju) [reply game_objects/ball.js:26]: Ok, it is when kicked.

**Pull request merged**